src/utils/firestore.js

```diff
@@ -40,130 +46,104 @@ const getDocsFromCollection = async (collectionName, constraints = []) => {
+            console.log('Updated entry');
+            return true;
         })
         .catch((e) => {
-            notify(Type.error, `Error deleting document: ${e}`);
+            console.log(`Error updating entry: ${e}`);
+            return false;
         });
-    if (entrySnapshot.data().taxa === 'Lizard') updateLizardMetadata('delete', { entrySnapshot });
 };
 
-const updateLizardMetadata = async (operation, operationDataObject) => {
-    if (operation === 'create') {
-    } else if (operation === 'update') {
-        await updateDoc(doc(db, 'Metadata', 'LizardData'), {
-            lastEditTime: operationDataObject.lastEditTime,
-        })
+const deleteEntry = async (entry) => {
+    const collectionName = getDocCollectionName(entry);
+    const isSession = collectionName.includes('Session');
+    const isLizard = entry.data().taxa === 'Lizard';
+    if (isLizard) {
+        await addDeletedLizardRecord(entry)
             .then(() => {
-                notify(Type.success, 'Sent update to the PWA');
+                console.log('Added deleted lizard record');
             })
             .catch((e) => {
-                notify(Type.error, `Error sending deletion to PWA: ${e}`);
-            });
-    } else if (operation === 'delete') {
-        const { entrySnapshot } = operationDataObject;
-        await updateDoc(doc(db, 'Metadata', 'LizardData'), {
-            deletedEntries: arrayUnion({
-                entryId: entrySnapshot.id,
-                collectionId: entrySnapshot.ref.parent.id,
-            }),
-        })
-            .then(() => {
-                notify(Type.success, 'Sent deletion to the PWA');
-            })
-            .catch((e) => {
-                notify(Type.error, `Error sending deletion to PWA: ${e}`);
+                console.log(`Error adding deleted lizard record: ${e}`);
+                return false;
             });
     }
-};
-
-const pushEntryChangesToFirestore = async (entrySnapshot, entryData) => {
-    if (entryData.taxa === 'Lizard') {
-        const lastEditTime = new Date().getTime();
-        entryData.lastEdit = lastEditTime;
-        updateLizardMetadata('update', { lastEditTime });
-    }
-    await setDoc(doc(db, entrySnapshot.ref.parent.id, entrySnapshot.id), entryData)
+    return deleteDoc(doc(db, collectionName, entry.id))
         .then(() => {
-            notify(Type.success, 'Changes successfully written to database!');
+            console.log('Deleted entry');
+            if (isSession) {
+                return deleteSessionEntries(entry)
+                    .then(() => {
+                        console.log('Deleted session entries');
+                        return true;
+                    })
+                    .catch((e) => {
+                        console.log(`Error deleting session entries: ${e}`);
+                        return false;
+                    });
+            }
+            return true;
```

Comment:
considering your last bullet point, this could return something along the lines of
```suggestion
            return [true, 'success message!'];
```

Reply:
Implemented